lua/telescope/utils.lua

```diff
@@ -446,7 +446,9 @@ utils.transform_devicons = load_once(function()
         return display
       end
 
-      local icon, icon_highlight = devicons.get_icon(utils.path_tail(filename), nil, { default = true })
+      local basename = utils.path_tail(filename)
+      -- the double :e:e gets multipart extensions, if one exists, like *.test.js, for example
+      local icon, icon_highlight = devicons.get_icon(basename, vim.fn.fnamemodify(basename, ":e:e"), { default = true })
```

Comment:
If we add something like this:
```
local icon, icon_highlight = devicons.get_icon(basename, basename:match("[^a-z][^A-Z]*$"), { default = false })
if icon == nil then
  icon, icon_highlight = devicons.get_icon(basename, vim.fn.fnamemodify(basename, ":e:e"), { default = true })
end
```
(Not a good solution, but just to get the point), we could have different icons for files names like these: `ControllerTests.java` and `Controller.java`. (Notice, that `Tests.java` in `ControllerTests.java` does not have a preceding dot, which means if there is an icon mapped to the name `Tests.java`, it would not get picked up with the current changes)
I think this would be cool to add.

Reply:
How would we know what the "extension" is in that case? Also we'll be looking to remove `fnamemodify` as per @Conni2461 's comment

Reply:
I don't know if I understand you correctly, but in nvim-devicons you can set the icons for a file extension like so:
```lua
require "nvim-web-devicons".set_icon {
  "spec.ts" = { icon = "A", color = "#00ffff", name = "AngularTest" }
}
```
This would set the icon for all `.spec.ts` files.
With my change however you can add another key like so:
```lua
require "nvim-web-devicons".set_icon {
  "spec.ts" = { icon = "A", color = "#00ffff", name = "AngularTest" }
  "Events.spec.ts" = { icon = "B", color = "#00ffff", name = "AngularExampleFileTest" }
}
```
And this would set the icon `B` for every file that is named `somethingEvents.spec.ts`. (notice, there is no dot preceding Events.spec.ts)

This is useful in languages like java, where test classes end with `Tests.java`, and NOT `.Tests.java`.  (Again, notice the preceding dot)
The code would look like this:

This would set the icon for all `.spec.ts` files. With my change however you can add another key like so:
```lua
require "nvim-web-devicons".set_icon {
  "java" = { icon = "J", color = "#00ffff", name = "Java" }
  "Tests.java" = { icon = "T", color = "#00ffff", name = "JavaTests" }
}
```
With my changes applied, it would have the icon T for all files ending with `Tests.java`, and the icon j for all files ending with `.java` (the dot)